app/objects/impersonation/authenticator.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module Impersonation
+  class Authenticator
+    def initialize(encrypted_data)
+      @encrypted_data = encrypted_data
+    end
+
+    def build_auth_headers!
+      user.tokens.filter! { |_token, attrs| attrs['impersonated_by'] != admin_user_id }
```

Comment:
```suggestion
      user.tokens.reject! { |_token, attrs| attrs['impersonated_by'] == admin_user_id }
```
really minor comment, feel free to ignore :P